couchdb/src/test/java/org/hibernate/ogm/datastore/couchdb/test/crud/EntityWithByteArrayTest.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Hibernate OGM, Domain model persistence for NoSQL datastores
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.ogm.datastore.couchdb.test.crud;
+
+import java.util.Arrays;
+import java.util.Random;
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.Id;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+import org.hibernate.annotations.GenericGenerator;
+import org.hibernate.ogm.utils.OgmTestCase;
+import org.hibernate.ogm.utils.TestForIssue;
+
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertNotNull;
+
+/**
+ * @author Hardy Ferentschik
+ */
+public class EntityWithByteArrayTest extends OgmTestCase {
```

Comment:
> It did not fail which was another reason I did not see it. Seems in the test the byte[] is mapped as a @Lob as well, so I guess that makes the difference. I think the mapping without @Lob should be valid as well

As you said, the test didn't fail because the use case is different from the one we are testing

> Looking at BuiltInTypeTest I have to say I am not a big fan of this test all in one go tests. I find them hard to understand and maintain. I am much more for small dedicated tests

Yes, BuiltInTypeTest.testTypesSupport is pretty big and I think it would be nice to refactor it.
We could probably separte each assert in it's own unit test.
That said, I think for the issue you are working at the moment it would be better to be consistent with what we have now.

> The test I wrote just addresses the actual JIRA issue in a minimal test case. Since this was a CouchDB test I added the test case there. I guess one could argue that a backendtck test for byte[] would make sense as well, but I see that as an add on to this dedicated test

The user discovered the exception using CouchDB but we should check and support built-in types on all the other datastores as well. It's very likely that it will work out of the box. If it doesn't it should be easy to fix. We have several dedicated test in in the project but I don't think this is a good use case for one.